### PR TITLE
[DUOS-1779][risk=no] Update Confirmation Modal Styling

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1779

## Changes

* Minor updates to the confirmation dialog to use the current button styles for all of our console pages.
* Updated usages of the confirmation dialog where necessary to tweak for the new fonts/buttons/etc.
* Also addressed minor bug fixes found during review.

### New: 

Screen Shot 2022-06-01 at 2 14 52 PM

Screen Shot 2022-06-01 at 2 17 23 PM

### Old

Screen Shot 2022-06-01 at 2 16 28 PM

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by shaemarks — COMMENTED** [review]
When I was looking at the different places the Confirmation Modal is used, I noticed that the Delete Draft DAR Modal from the Researcher Console has the incorrect title (see image below), and that Deactivate Library Card Modal on the Signing Official Console is taller than expected. Both of these bugs seem to be from before this PR, but I thought I would mention it. Changes look good otherwise

Delete Draft Modal

**Comment by rushtong** [comment]
> When I was looking at the different places the Confirmation Modal is used, I noticed that the Delete Draft DAR Modal from the Researcher Console has the incorrect title (see image below), and that Deactivate Library Card Modal on the Signing Official Console is taller than expected. Both of these bugs seem to be from before this PR, but I thought I would mention it. Changes look good otherwise

Nice catches - I looked at these usages too and missed those bugs. Will address in this PR.

**Comment by rushtong** [comment]
@shaemarks - Updated, PTAL, thank you!

File: src/components/dar_drafts_table/DarDraftTable.js [hunk 1]
```diff
@@ -243,11 +243,10 @@ export default function DarDraftTable(props) {
     }),
     h(ConfirmationModal, {
       showConfirmation,
-      styleOverrise: { height: '35%' },
       closeConfirmation: () => setShowConfirmation(false),
       title: 'Delete Draft DAR',
       message: `Are you sure you want to delete DAR draft ${getIdentifier({id: selectedDraft.id, data: selectedDraft.data})}`,
-      header: getModalHeader(),
+      header: getModalHeader,
```
Comment (shaemarks) [inline src/components/dar_drafts_table/DarDraftTable.js]: I'm realizing now that the example I was looking at with "Title" as the header was actually for a DAR draft named "Title" (sorry about that!). Now I'm not seeing a header at all, so I think the original code was correctly making the header the name of the draft

Comment (rushtong) [reply src/components/dar_drafts_table/DarDraftTable.js]: Hah, will revert that change then!

Comment (rushtong) [reply src/components/dar_drafts_table/DarDraftTable.js]: There was a second bug on that page where it was not finding the partial dar code correctly. Also fixed that.

File: src/components/dar_drafts_table/DarDraftTable.js [hunk 2]
```diff
@@ -97,7 +97,7 @@ const ResumeDraftButton = (props) => {
 
 //helper function to create id string for error messages
 const getIdentifier = ({id, data}) => {
-  return !isEmpty(data) ? (data.projectTitle || data.tempDarCode) : id;
+  return !isEmpty(data) ? (data.projectTitle || data.partialDarCode) : id;
```
Comment (rushtong) [inline src/components/dar_drafts_table/DarDraftTable.js:100]: Existing bug found during review.

**Review by shaemarks — COMMENTED** [review]
The modals used in the `DarCollectionTable` currently don't have headers because the
 `header: getModalHeader` lines need parentheses after them. See line 38 of `CollectionConfirmationModal` as an example.

**Comment by rushtong** [comment]
> The modals used in the DarCollectionTable currently don't have headers because the
header: getModalHeader lines need parentheses after them. See line 38 of CollectionConfirmationModal as an example.

Updated, thank you!

**Review by shaemarks — APPROVED** [review]
Looks good! Thanks for the changes!

File: src/components/dar_collection_table/CollectionConfirmationModal.js [hunk 3]
```diff
@@ -32,11 +32,10 @@ export default function CollectionConfirmationModal(props) {
   const cancelModal =
     h(ConfirmationModal, {
       showConfirmation,
-      styleOverride: {height: '35%'},
       closeConfirmation: () => setShowConfirmation(false),
       title: 'Cancel DAR Collection',
       message: `Are you sure you want to cancel ${collection.darCode}?`,
-      header: getModalHeader,
+      header: getModalHeader(),
```
Comment (rushtong) [inline src/components/dar_collection_table/CollectionConfirmationModal.js:38]: Existing bug found during review.

File: src/components/dar_collection_table/CollectionConfirmationModal.js [hunk 4]
```diff
@@ -47,17 +46,16 @@ export default function CollectionConfirmationModal(props) {
       closeConfirmation: () => setShowConfirmation(false),
       title: 'Revise DAR Collection',
       message: `Are you sure you want to revise ${collection.darCode}?`,
-      header: getModalHeader,
+      header: getModalHeader(),
```
Comment (rushtong) [inline src/components/dar_collection_table/CollectionConfirmationModal.js:49]: Existing bug found during review.

File: src/components/dar_collection_table/CollectionConfirmationModal.js [hunk 5]
```diff
@@ -47,17 +46,16 @@ export default function CollectionConfirmationModal(props) {
...
     });
 
   const openModal = h(ConfirmationModal, {
     showConfirmation,
-    styleOverride: { height: '35%' },
     closeConfirmation: () => setShowConfirmation(false),
     title: 'Open DAR Collection',
     message: `Are you sure you want to open ${collection.darCode}?`,
-    header: getModalHeader,
+    header: getModalHeader(),
```
Comment (rushtong) [inline src/components/dar_collection_table/CollectionConfirmationModal.js:58]: Existing bug found during review.

File: src/components/modals/ConfirmationModal.js [hunk 6]
```diff
@@ -1,37 +1,64 @@
...
 const ConfirmationModal = (props) => {
   const {showConfirmation, closeConfirmation, title, message, header, onConfirm, styleOverride = {}} = props;
   const closeFn = () => closeConfirmation();
 
-  const confirmButton = button({className: 'cell-button hover-color', onClick: onConfirm}, ['Confirm']);
-  const computedStyle = Object.assign({}, Styles.MODAL.CONFIRMATION, styleOverride);
+  const duosBlue = 'rgb(0, 96, 159)';
+  const duosBlueHover = 'rgb(9,72,183)';
+
+  const SecondaryButton = styled(Button)(() => ({
```
Comment (lu-c) [inline src/components/modals/ConfirmationModal.js:17]: I think these styles would be great candidates for the global styles declaration, however we choose to handle those in the future! :)

Comment (rushtong) [reply src/components/modals/ConfirmationModal.js:17]: Agreed - I was pulling these out explicitly for that reason.

File: src/components/modals/ConfirmationModal.js [hunk 7]
```diff
@@ -1,37 +1,64 @@
...
+    },
+  }));
+
+  const PrimaryButton = styled(Button)(({theme}) => ({
+    fontFamily: 'Montserrat, sans-serif',
+    color: theme.palette.getContrastText(duosBlue),
+    backgroundColor: duosBlue,
+    borderRadius: '4px',
+    fontSize: '1.45rem',
+    '&:hover': {
```
Comment (lu-c) [inline src/components/modals/ConfirmationModal.js:36]: this really is useful, and addresses some of my concerns about using js styles ::)

Comment (rushtong) [reply src/components/modals/ConfirmationModal.js:36]: I wasn't able to use regular css with the `Button` components ... had to go this route.

**Review by lu-c — APPROVED** [review]
LGTM

**Pull request merged**